src/test/app/Check_test.cpp

```diff
@@ -1848,22 +1883,654 @@ class Check_test : public beast::unit_test::suite
+
+            // Use check cashing to automatically create the trust line.
+            IOU const CK5 = gw2["CK5"];
+            uint256 const chkId{getCheckIndex(gw2, env.seq(gw2))};
+            env(check::create(gw2, alice, CK5(92)));
+            env.close();
+            BEAST_EXPECT(
+                env.le(keylet::line(gw2, alice, CK5.currency)) == nullptr);
+            env(check::cash(alice, chkId, CK5(92)), ter(tecNO_AUTH));
+            env.close();
+
+            // No one's owner count should have changed.
+            BEAST_EXPECT(ownerCount(env, gw1) == 0);
+            BEAST_EXPECT(ownerCount(env, alice) == 6);
+            BEAST_EXPECT(ownerCount(env, bob) == 5);
+
+            // Because gw2 has set lsfRequireAuth, neither trust line
+            // is created.
+            BEAST_EXPECT(
+                env.le(keylet::line(gw2, alice, OF5.currency)) == nullptr);
+            BEAST_EXPECT(
+                env.le(keylet::line(gw2, alice, CK5.currency)) == nullptr);
+        }
+        //------------ lsfRequireAuth, check written by non-issuer -------------
+        {
+            // Since gw2 has the lsfRequireAuth flag set, there should be
+            // no automatic trust line creation between non-issuers.
+
+            // Use offers to automatically create the trust line.
+            IOU const OF5 = gw2["OF5"];
+            env(offer(alice, XRP(91), OF5(91)), ter(tecUNFUNDED_OFFER));
+            env.close();
+            env(offer(bob, OF5(91), XRP(91)), ter(tecNO_LINE));
+            BEAST_EXPECT(
+                env.le(keylet::line(gw2, bob, OF5.currency)) == nullptr);
+            env.close();
+
+            BEAST_EXPECT(ownerCount(env, gw1) == 0);
+            BEAST_EXPECT(ownerCount(env, alice) == 6);
+            BEAST_EXPECT(ownerCount(env, bob) == 5);
+
+            // Use check cashing to automatically create the trust line.
+            IOU const CK5 = gw2["CK5"];
+            uint256 const chkId{getCheckIndex(alice, env.seq(alice))};
+            env(check::create(alice, bob, CK5(91)));
+            env.close();
+            BEAST_EXPECT(
+                env.le(keylet::line(alice, bob, CK5.currency)) == nullptr);
+            env(check::cash(bob, chkId, CK5(91)), ter(tecPATH_PARTIAL));
+            env.close();
+
+            // Delete alice's check since it is no longer needed.
+            env(check::cancel(alice, chkId));
+            env.close();
+
+            // No one's owner count should have changed.
+            BEAST_EXPECT(ownerCount(env, gw1) == 0);
+            BEAST_EXPECT(ownerCount(env, alice) == 6);
+            BEAST_EXPECT(ownerCount(env, bob) == 5);
+
+            // Because gw2 has set lsfRequireAuth, neither trust line
+            // is created.
+            BEAST_EXPECT(
+                env.le(keylet::line(gw2, bob, OF5.currency)) == nullptr);
+            BEAST_EXPECT(
+                env.le(keylet::line(gw2, bob, CK5.currency)) == nullptr);
+        }
```

Comment:
I think this could use another set of cases where the authorized trust lines are created. Obviously, there won't be any trustlines created automatically, but it can still test going over the trust line limit, and verify things still work with the `asfRequireAuth` flag.